### PR TITLE
Interactive  + Self-chat + Pretty display docs

### PR DESCRIPTION
**Pull request by jaseweston**
Some more docs!! (see title)

### CODE REVIEW

File: docs/source/tutorial_tipsntricks.rst [hunk 1]
```diff
@@ -44,6 +44,39 @@ That is, by adding a colon ":" followed by the flag name, an equals sign, and th
 You can add multiple flags, all separated by ":".
 
 
+Self-Chats
+##########
+
+Sometimes it is useful to generate models talking to themselves. You can do this with:
```
Comment (klshuster) [inline docs/source/tutorial_tipsntricks.rst]: could you maybe elaborate a bit on why you would specify something like `convai2:selfchat`? and perhaps include other tasks for which this is a thing?

File: docs/source/tutorial_tipsntricks.rst [hunk 2]
```diff
@@ -44,6 +44,39 @@ That is, by adding a colon ":" followed by the flag name, an equals sign, and th
...
+.. code-block:: python
+
+   # Self-chatting Poly-Encoder model on a generic task (so e.g., no ConvAI2 personas are input)
+   python parlai/scripts/self_chat.py -mf zoo:pretrained_transformers/model_poly/model -t self_chat --inference topk -ne 10 --display-examples True -dt valid
+
+
+Prettifying Display of Chats
+############################
+
+This handy script can prettify the display of json file of chats (sequences of parlai messages):
```
Comment (klshuster) [inline docs/source/tutorial_tipsntricks.rst]: does the data need to be in a certain format to be prettified?

Comment (jaseweston) [reply docs/source/tutorial_tipsntricks.rst]: it does but i think we need to fix that to be more permissive of any json of parlai messages, so i didn't really want to document the current given that it should be fixed. what i do give are example files so they can be looked at..

File: docs/source/tutorial_tipsntricks.rst (hunk 1 above)
Comment (jaseweston) [reply docs/source/tutorial_tipsntricks.rst]: I have written that below:

The task set by '-t' (in the above case "convai2:selfchat") links to a parlAI world that handles the particular nature of interactions, see e.g. `here <https://github.com/facebookresearch/ParlAI/blob/master/parlai/tasks/convai2/worlds.py#L98>`_ 
or `here <https://github.com/facebookresearch/ParlAI/blob/master/parlai/tasks/wizard_of_wikipedia/worlds.py#L106>`_.

Comment (klshuster) [reply docs/source/tutorial_tipsntricks.rst]: not sure how i missed that, sorry

File: docs/source/tutorial_tipsntricks.rst (hunk 2 above)
Comment (klshuster) [reply docs/source/tutorial_tipsntricks.rst]: works for me

File: docs/source/tutorial_tipsntricks.rst (hunk 1 above)
Comment (jaseweston) [reply docs/source/tutorial_tipsntricks.rst]: well it was kind of there, but i updated it also!

**Pull request merged**